integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/steps/Hooks.java

```diff
@@ -36,20 +43,40 @@ public void before(Scenario scenario) {
 
     @After
     public void cleanUp() {
-        String token = BridgeUtils.retrieveBridgeToken();
-        String bridgeId = context.getBridgeId();
-        if (BridgeResource.getBridgeList(token).getItems().stream()
-                .anyMatch(b -> b.getId().equals(context.getBridgeId()))) {
-            BridgeResponse bridge = BridgeResource.getBridgeDetails(context.getManagerToken(), bridgeId);
-            if (bridge.getStatus() == BridgeStatus.READY) {
-                if (ProcessorResource.getProcessorList(token, bridgeId).getSize() > 0) {
-                    ProcessorResource.getProcessorList(token, bridgeId).getItems().stream()
-                            .forEach(p -> ProcessorResource.deleteProcessor(token, bridgeId, p.getId()));
-                    Awaitility.await().atMost(Duration.ofMinutes(2)).pollInterval(Duration.ofSeconds(5))
-                            .until(() -> ProcessorResource.getProcessorList(token, bridgeId).getSize() == 0);
-                }
-            }
-            BridgeResource.deleteBridge(token, bridgeId);
-        }
+        // Remove all bridges/processors created
+        context.getAllBridges().values()
+                .stream()
+                .filter(bridgeContext -> !bridgeContext.isDeleted())
+                .forEach(bridgeContext -> {
+                    final String bridgeId = bridgeContext.getId();
+                    BridgeResponse bridge = BridgeResource.getBridgeDetails(context.getManagerToken(), bridgeId);
+                    if (bridge.getStatus() == BridgeStatus.READY) {
+                        ProcessorListResponse processorList = ProcessorResource.getProcessorList(
+                                context.getManagerToken(),
+                                bridgeId);
+                        if (processorList.getSize() > 0) {
+                            processorList.getItems().stream().forEach(
+                                    p -> ProcessorResource.deleteProcessor(context.getManagerToken(), bridgeId,
+                                            p.getId()));
+                            Awaitility.await()
+                                    .atMost(Duration.ofMinutes(2))
+                                    .pollInterval(Duration.ofSeconds(5))
+                                    .until(() -> ProcessorResource.getProcessorList(context.getManagerToken(), bridgeId)
+                                            .getSize() == 0);
+                        }
+                    }
+                    switch (bridge.getStatus()) {
+                        case ACCEPTED:
+                        case PROVISIONING:
```

Comment:
For these cases IMHO the Bridge should be deleted too.
Also FAILED state should be added and hook should delete bridge.

Reply:
The bridge will be deleted if `ready`, `accepted` or `provisioning`
I will add the `failed`